src/components/MyNyplButton/MyNyplButton.js

```diff
@@ -81,7 +83,11 @@ class MyNyplButton extends React.Component {
     // going to the link
     e.preventDefault();
     const visibleState = this.state.visible ? 'Closed' : 'Open';
-
+    setTimeout(() => {
```

Comment:
why do you need a setTimeout here?

Reply:
In the image below, the "You are logged in..." part should be focused when the dropdown is opened. Since this is dynamically created when the component mounts (making a call to fetch patron info), it, for some reason, could not pick up that the DOM was updated and did not want to focus correctly. The `setTimeout` allows it to focus on the proper element.

![screen shot 2017-12-14 at 2 28 17 pm](https://user-images.githubusercontent.com/1280564/34010518-090538d2-e0db-11e7-8e7b-9c8ee5300458.png)

Reply:
That works, but seems like focus could be set by the element itself in MyNypl.componentDidMount ? e.g.

```  
class MyNypl extends React.Component {
  componentDidMount() {
    if (this.greetingDiv) this.greetingDiv.focus();
    else this.refs.catalogLink.focus();
  }
  ...
  renderGreeting() {
    ...
    return (
      <div tabIndex="0" className="patron-greeting-wrapper" ref={(el) => {
          this.greetingDiv = el; 
        }}  
      >  
```

Maybe that approach wouldn't work here though.

Reply:
I'll try it.

Reply:
It works, and it's better than having a setTimeout. Thanks!